apps/04TowersOfHanoi.js

```diff
@@ -19,24 +19,47 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+// Removes the last item of the startStack array,
+// and places it on the end of the endStack array
+function movePiece(startStack, endStack) {
+  var pieceBeingMoved = stacks[startStack].pop();
+  stacks[endStack].push(pieceBeingMoved);
 }
 
-function isLegal() {
-  // Your code here
-
+// Does not allow larger blocks to be placed on top of
+// smaller blocks; Also permits any block to be placed on
+// an empty stack
+function isLegal(startStack, endStack) {
+  var startBlock = stacks[startStack][(stacks[startStack].length - 1)];
+  var endBlock = stacks[endStack][(stacks[endStack].length - 1)];
+
+  if (startBlock < endBlock) {
+    return true;
+  }
+  if (stacks[endStack].length === 0) {
```

Comment:
What about if startStack's length is 0?

Also, should we perhaps move the length check to BEFORE we do the value comparison on line 36?